File: src/main/java/no/nav/sbl/dialogarena/mininnboks/consumer/TekstServiceImpl.java
package no.nav.sbl.dialogarena.mininnboks.consumer;

import lombok.SneakyThrows;
import org.apache.commons.io.FileUtils;

import java.io.File;
import java.net.URL;
import java.util.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import static java.util.Optional.ofNullable;

public class TekstServiceImpl implements TekstService {

    private final Map<String, String> tekster = new HashMap<>();

    public TekstServiceImpl() {
        lastTekster();
    }

    @SneakyThrows
    private void lastTekster() {
        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
        Enumeration<URL> resources = contextClassLoader.getResources("tekster/mininnboks");
//        Enumeration<URL> resources2 = contextClassLoader.getResources("tekster/mininnboks/svar_nb.txt");

        Collections.list(resources).stream().map(URL::getFile).map(File::new).forEach(this::lastTekster);

        Collections.list(resources).forEach(r -> {
            System.out.println(r);
        });
    }

    @SneakyThrows
    private void lastTekster(File file) {
        if (file.isDirectory()) {
            Arrays.stream(file.listFiles()).forEach(this::lastTekster);
        } else {
            String name = file.getName();
            Pattern compile = Pattern.compile("(.*)_nb.(txt|html)");
            Matcher matcher = compile.matcher(name);
            if (!matcher.find()) {
                throw new IllegalStateException(name);
            }
            String substring = matcher.group(1);
            tekster.put(substring, FileUtils.readFileToString(file, "UTF-8").trim());
        }
    }

    @Override
    public String hentTekst(String key) {
        return ofNullable(tekster.get(key)).orElseThrow(() -> new MissingResourceException("mangler tekst for key=" + key, key, key));
    }

    @Override
    public Map<String, String> hentTekster() {
        return tekster;
    }
}

File: src/main/java/no/nav/sbl/dialogarena/mininnboks/consumer/TekstServiceImpl.kt
package no.nav.sbl.dialogarena.mininnboks.consumer

import lombok.SneakyThrows
import org.apache.cxf.helpers.FileUtils
import java.io.File
import java.net.URL
import java.util.*
import java.util.regex.Pattern

class TekstServiceImpl : TekstService {
    private val tekster: MutableMap<String?, String?> = HashMap()

    @SneakyThrows
    private fun lastTekster() {
        val resources = TekstServiceImpl::class.java.classLoader.getResources("tekster/mininnboks")
        Collections.list(resources).stream().map { obj: URL -> obj.file }.map { pathname: String? -> File(pathname) }.forEach { file: File -> this.lastTekster(file) }
    }

    @SneakyThrows
    private fun lastTekster(file: File) {
        if (file.isDirectory) {
            Arrays.stream(file.listFiles()).forEach { file: File -> this.lastTekster(file) }
        } else {
            tekster[finnKey(file)] = FileUtils.getStringFromFile(file).trim { it <= ' ' }
        }
    }

    private fun finnKey(tekstFil: File): String {
        val fileName = tekstFil.name
        val matcher = Pattern.compile("(.*)_nb.(txt|html)").matcher(fileName)
        check(matcher.find()) { fileName }
        return matcher.group(1)
    }

    override fun hentTekst(key: String?): String? {
        return Optional.ofNullable(tekster[key]).orElseThrow { MissingResourceException("mangler tekst for key=$key", key, key) }
    }

    override fun hentTekster(): Map<String?, String?>? {
        return tekster
    }

    init {
        lastTekster()
    }
}
